lib/database/set-claim.php

```diff
@@ -0,0 +1,500 @@
+ */
+function extendClaim(string $user, int $gameID): bool
+{
+    sanitize_sql_inputs($gameID);
+
+    if (hasSetClaimed($user, $gameID, true)) {
+        $query = "
+            UPDATE
+                SetClaim
+            SET
+                Extension = Extension + 1,
+                Finished = DATE_ADD(Finished, INTERVAL 3 MONTH),
+                Updated = NOW()
+            WHERE
+                Status = 0
+                AND GameID = '$gameID'
+                AND TIMESTAMPDIFF(MINUTE, NOW(), Finished) <= 10080";
+
+        if (s_mysql_query($query)) {
+            return true;
+        }
+    }
+    return false;
+}
+
+/**
+ * Gets the claim data for a specific game to display to the users.
+ */
+function getClaimData(int $gameID, bool $getFullData = true): array
+{
+    sanitize_sql_inputs($gameID);
+
+    $retVal = [];
+    if ($getFullData) {
+        $query = "
+        SELECT
+            sc.User as User,
+            sc.SetType as SetType,
+            sc.ClaimType as ClaimType,
+            sc.Created as Created,
+            sc.Finished as Expiration,
+            sc.Status as Status,
+            TIMESTAMPDIFF(MINUTE, NOW(), sc.Finished) AS MinutesLeft,
+            TIMESTAMPDIFF(MINUTE, sc.Created, NOW()) AS MinutesActive";
+    } else {
+        $query = "
+        SELECT
+            sc.User as User,
+            sc.SetType as SetType,
+            sc.ClaimType as ClaimType,
+            sc.Created as Created,
+            sc.Finished as Expiration";
+    }
+    $query .= "
+        FROM
+            SetClaim sc
+        WHERE
+            sc.GameID = '$gameID'
+            AND sc.Status = 0
+        ORDER BY
+            sc.ClaimType ASC";
+
+    $dbResult = s_mysql_query($query);
+
+    if ($dbResult !== false) {
+        while ($nextData = mysqli_fetch_assoc($dbResult)) {
+            $retVal[] = $nextData;
+        }
+    }
+    return $retVal;
+}
+
+/**
+ * Gets all the filtered claim list data.
+ * This includes user, game information, claim type, set type, claim status and claim timestamps.
+ *
+ * Results are configurable based on input parameters, allowing sorting on each of the
+ * above stats and returning data for a specific user or game.
+ */
+function getFilteredClaimData(int $gameID = 0, int $claimFilter = 511, int $sortType = 8, bool $getExpiringOnly = false, ?string $username = null, bool $getCount = false, int $offset = 0, int $limit = 50): array|int
+{
+    $retVal = [];
+    sanitize_sql_inputs($gameID, $username);
+
+    $primaryClaim = ($claimFilter & (1 << 0));
+    $collaborationClaim = ($claimFilter & (1 << 1));
+    $newSetClaim = ($claimFilter & (1 << 2));
+    $revisionClaim = ($claimFilter & (1 << 3));
+    $activeClaim = ($claimFilter & (1 << 4));
+    $completeClaim = ($claimFilter & (1 << 5));
+    $droppedClaim = ($claimFilter & (1 << 6));
+    $developerClaim = ($claimFilter & (1 << 7));
+    $juniorDeveloperClaim = ($claimFilter & (1 << 8));
```

Comment:
Please create an enum for this. See `TicketFilters`

Reply:
Updated to use enums.